### PR TITLE
Upgrade to Cypress v12

### LINKED ISSUE #8055: Upgrade to Cypress v12

### PR DESCRIPTION
**Pull request by jpandersen87**
Fixes #8055

This PR updates cypress to v12 and updates related dependencies to their latest versions. ESLint rules were modified (as suggested here: **https://github.com/testing-library/eslint-plugin-testing-library/discussions/533#discussioncomment-1949661**) to not use testing-library rules on cypress files to fix false-positives. ESLint plugin chai-friendly was added to cypress ruleset to allow unused-expressions on chai method chains.

### CODE REVIEW

File: frontend-react/cypress.config.ts [hunk 1]
```diff
@@ -12,9 +13,36 @@ const cypressConfig = require(path.resolve(
     `./cypress/cypress.${cypressEnv}.json`
 ));
 
+function getEnvOrDefault(name: string, dfault?: string) {
```
Comment (stephenkao) [inline frontend-react/cypress.config.ts:16]: Ah, minor typo: `default`

Comment (jpandersen87) [reply frontend-react/cypress.config.ts:16]: default is a reserved word 😛

Comment (stephenkao) [reply frontend-react/cypress.config.ts:16]: Hahaha oh duh.  Can we make it something like `defaultValue` instead then?

File: frontend-react/cypress/support/commands/auth.ts [hunk 2]
```diff
@@ -17,9 +17,10 @@ const SECRET = Cypress.env("oktaSecret");
 const CLIENT_ID = Cypress.env("oktaClientId");
 const REDIRECT_URI = `${Cypress.env("baseUrl")}/login/callback`;
 const RESPONSE_MODE = "okta_post_message";
+const URL = Cypress.env("oktaUrl");
 
 const oktaAuthConfig: OktaAuthOptions = {
-    issuer: "https://hhs-prime.oktapreview.com/oauth2/default",
+    issuer: `${URL}/oauth2/default`,
```
Comment (stephenkao) [inline frontend-react/cypress/support/commands/auth.ts:23]: Thanks for this cleanup!

**Review by stephenkao — APPROVED** [review]
Looks and works great!  Thanks so much for hopping on this!

**Pull request merged**